hpx/parallel/util/scan_partitioner.hpp

```diff
@@ -349,19 +349,6 @@ namespace hpx { namespace parallel { namespace util
                                 f1, f2, f3, f4);
                     });
 
-                if (f.has_exception())
-                {
-                    try {
-                        std::rethrow_exception(f.get_exception_ptr());
-                    }
-                    catch (std::bad_alloc const& ba) {
-                        throw ba;
-                    }
-                    catch (...) {
-                        return hpx::make_exceptional_future<R>(std::current_exception());
-                    }
-                }
-
```

Comment:
Could you elaborate on why this has to be removed please?

Reply:
@sithhell This is fundamentally wrong.
```f.has_exception()``` always returns ```false``` when the task is running.
And, even if ```f.has_exception()``` can return ```true```, this code is still wrong because occurred exception must be wrapped into future.

Reply:
thanks for the clarification. makes sense.